packages/core/ios/RNEmbrace/EmbraceManager.swift

```diff
@@ -0,0 +1,301 @@
+import EmbraceCore
+import EmbraceCrash
+import EmbraceCommonInternal
+
+#if canImport(CodePush)
+import CodePush
+#endif
+
+enum EmbraceKeys: String {
+  case reactNativeVersion = "io.embrace.reactnative.version"
+  case embraceReactNativeSdkVersion = "io.embrace.reactnative.sdk.version"
+  case javaScriptPatchNumber = "io.embrace.javascript.patch"
+  case javaScriptBundleURL = "io.embrace.jsbundle.url"
+}
+
+@objc(EmbraceManager)
+class EmbraceManager: NSObject {
+  @objc
+  func setJavaScriptBundlePath(_ path: String, resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
+    do {
+      try Embrace.client?.metadata.addResource(key: EmbraceKeys.javaScriptBundleURL.rawValue, value: path)
+      resolve(true)
+    }  catch let error {
+      reject("SET_JS_BUNDLE_PATH_ERROR", "Error setting JavaScript bundle path", error)
+    }
+  }
+  
+  @objc
+  func isStarted(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
+    if let embraceStarted = Embrace.client?.started {
+      resolve(embraceStarted)
+    } else {
+      resolve(false)
+    }
+  }
+  
+  @objc(startNativeEmbraceSDK:resolver:rejecter:)
+  func startNativeEmbraceSDK(_ appId: String, resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
+    DispatchQueue.main.async {
+      do {
+        
+        var embraceOptions: Embrace.Options {
+          return .init(
+            appId: appId,
+            appGroupId: nil,
+            platform: .reactNative,
+            captureServices: .automatic,
+            crashReporter: EmbraceCrashReporter()
+          )
+        }
+        
+        try Embrace.setup(options: embraceOptions)
+          .start()
+        
+        resolve(true)
+      } catch let error {
+        reject("START_EMBRACE_SDK", "Error starting Embrace SDK", error)
+      }
+    }
+  }
+  
+  @objc
+  func getDeviceId(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
+    if let deviceId = Embrace.client?.currentDeviceId() {
+      resolve(deviceId)
+    } else {
+      reject("GET_DEVICE_ID", "Error getting deviceId", nil)
+    }
+  }
+  
+  @objc
+  func setUserIdentifier(_ userIdentifier: String, resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
+    Embrace.client?.metadata.userIdentifier = userIdentifier
+    resolve(true)
+  }
+  
+  
+  @objc
+  func getCurrentSessionId(_ resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
+    if let sessionId = Embrace.client?.currentSessionId() {
+      resolve(sessionId)
+    } else {
+      reject("GET_SESSION_ID", "Error getting sessionId", nil)
+    }
+  }
+  
+  @objc
+  func addBreadcrumb(_ event: String, resolver resolve: @escaping RCTPromiseResolveBlock, rejecter reject: @escaping RCTPromiseRejectBlock) {
+    //TODO Refactor SPAN
```

Comment:
This should be something like
```swift
Embrace.client?.add(.breadcrumb(event)
resolve(true)
```

Reply:
noted in https://www.notion.so/embraceio/Implement-addBreadcrumb-in-iOS-native-module-8ec0d6f2e04d4b49a7b8fa7d669cf59b?pvs=4, can do it here if it's quick or punt to that ticket in the interest of wrapping up this PR

Reply:
Just providing context; if you guys prefer to do it in another PR, go ahead!